cpp/src/merge/merge.cu

```diff
@@ -45,8 +53,51 @@
 
 namespace cudf {
 namespace detail {
+
 namespace {
 
+template <bool has_nulls>
+struct row_lexicographic_tagged_comparator {
+  row_lexicographic_tagged_comparator(table_device_view const lhs,
+                                      table_device_view const rhs,
+                                      device_span<order const> const column_order,
+                                      device_span<null_order const> const null_precedence)
+    : _lhs{lhs}, _rhs{rhs}, _column_order{column_order}, _null_precedence{null_precedence}
+  {
+    // Add check for types to be the same.
```

Comment:
Is this comment a TODO for this PR?

Reply:
Oh it was already there before my changes, but now I realize it's a relevant comment still. I'll add fulfill this TODO.

Reply:
Okay looks like these checks were already performed in the call stack before this